com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ArchiveBuilder.java

```diff
@@ -310,39 +315,41 @@ private void writeArchiveIndex(File archiveIndexFile) throws IOException {
 
 
     /**
-     * Write the archive data and excluded resources to disk. This function will
+     * Write the archive entries to disk or archive file. This function will
      * create a thread pool and write the entries in separate threads to speed
-     * up the archive creation.
+     * up the process. The entries which are excluded will be put in one list
+     * while the included entries will be put in another.
      * @param archiveDataFile The archive data file to write to
      * @param resourcePackDirectory The directory where excluded resources will
      * be stored
-     * @param excludedResources List of resources which should be excluded from
-     * the archive
+     * @param excludedResources The resources to exclude from the archive
      */
-    private void writeArchiveData(File archiveDataFile, Path resourcePackDirectory, List<String> excludedResources) throws IOException, CompileExceptionError {
-        TimeProfiler.start("WriteArchiveData");
+    private void writeArchiveEntries(final File archiveDataFile, final Path resourcePackDirectory, List<String> excludedResources) throws IOException, CompileExceptionError {
+        TimeProfiler.start("writeArchiveEntries");
 
         // create the executor service to write entries in parallel
         int nThreads = project.getMaxCpuThreads();
-        logger.info("Creating archive data with a fixed thread pool executor using %d threads", nThreads);
+        logger.info("Creating archive entries with a fixed thread pool executor using %d threads", nThreads);
         this.executorService = Executors.newFixedThreadPool(nThreads);
         
         RandomAccessFile archiveData = new RandomAccessFile(archiveDataFile, "rw");
 
         Collections.sort(entries); // Since it has no hash, it sorts on path
 
+        excludedEntries = new ArrayList<>(entries.size());
+        includedEntries = new ArrayList<>(entries.size());
```

Comment:
Should both of them be entries.size()? 
Can we avoid creation of excludedEntries at all if build without liveupdate (no excluded entries)?